### DISCUSSION TITLE
How can i adjust the native Searchbar icon before headerRight elements ?

### DISCUSSION THREAD

**Discussion by arnelcariaga** [software-mansion/react-native-screens#1900] [General]
### Description

When i set this option to display the Searchbar it's show at the end of the header but i want to show the Searchbar before the headerRight elements and not after.

```
headerSearchBarOptions: {
                disableBackButtonOverride: true,
                obscureBackground: false,
                hideNavigationBar: true
}
```

Current screen
![249520409-1da076b1-929f-411a-a944-85b253ec897d](https://github.com/software-mansion/react-native-screens/assets/42476817/ae8e0a79-b628-4347-9443-451d0793c430)

### Steps to reproduce

Expected screen
![249520690-f568b9c3-68d0-419a-b719-60eeb3400162](https://github.com/software-mansion/react-native-screens/assets/42476817/a8be94c4-7e1d-435a-9395-1108d9000b3e)

### Snack or a link to a repository

https://snack.expo.dev/@arnelcariaga/c7790c

### Screens version

~3.20.0

### React Native version

0.71.8

### Platforms

Android

### JavaScript runtime

None

### Workflow

Expo bare workflow

### Architecture

None

### Build type

None

### Device

Real device

### Acknowledgements

Yes

**Comment by kkafar** [comment]
Hey, I do not believe it is possible with `screens`. I'm also moving this issue to discussions section of the repository.

**Comment by tboba** [comment]
Hi @arnelcariaga! As @kkafar said, for now it's not possible to do such thing in `react-native-screens` 🥲
That's because for the Search Bar we're using Options menu from Android API and that's the only type of option we're using on the native side - when you're using `headerRight` option we're just using gravity for the header subviews, so everything you're putting to the `headerRight` option is going behind the search bar icon, because this icon goes to the rightmost position of the header view.

We're planning to introduce an API for managing and adding more items to the options menu, but it's not our priority for now.
I'll close this discussion, but in case you've got any questions, don't hesitate to ask them!
